### PR TITLE
Adapt to upstream changes in the Checker Framework

### PR DESCRIPTION
**Pull request by bitterfox**
For PR https://github.com/typetools/checker-framework/pull/1390

### CODE REVIEW

File: src/checkers/inference/VariableAnnotator.java [hunk 1]
```diff
@@ -1172,8 +1172,12 @@ private void shallowAnnotateArray(AnnotatedTypeMirror type, Tree tree, int level
             addPrimaryVariable(type, tree);
         } else {
             TreePath pathToTopLevelTree = inferenceTypeFactory.getPath(topLevelTree);
-            ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree).newArrayLevel(level);
-            createEquivalentSlotConstraints(type, tree, new AstPathLocation(astRecord));
+            AstPathLocation location = null;
+            if (pathToTopLevelTree != null) {
+                ASTRecord astRecord = ASTPathUtil.getASTRecordForPath(inferenceTypeFactory, pathToTopLevelTree).newArrayLevel(level);
```
Comment (wmdietl) [inline src/checkers/inference/VariableAnnotator.java]: Can you add a comment explaining when this branch is taken?

Comment (wmdietl) [reply src/checkers/inference/VariableAnnotator.java]: As discussed, @bitterfox will add a short comment that for the artificial varargs trees we don't need an ASTPath.
After that change we can merge this.

**Comment by bitterfox** [comment]
Hi, I've updated this. Please check this again.

**Comment by wmdietl** [comment]
Thanks!

**Pull request merged**